### PR TITLE
[PART 2] VPN-5181 - Flatten /shared folder

### PR DESCRIPTION
**Pull request by brizental**
This is a reaplication of https://github.com/mozilla-mobile/mozilla-vpn-client/pull/7936 on top of main and the second (and final) part of the large restructuring under VPN-5181.

Full disclosure I did change a bit the approach from that PR. Essentially I keep the shared-sources cmake file to make transition easier.

### CODE REVIEW

**Comment by brizental** [comment]
Probe scraper failure needs to be fixed upstream as well. Sending a PR...

File: src/cmake/shared-sources.cmake [hunk 1]
```diff
@@ -0,0 +1,268 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Library of sources shared between app and tests
+add_library(shared-sources INTERFACE)
```
Comment (Gela) [inline src/cmake/shared-sources.cmake:6]: Why do we have a shared cmake and sources.cmake as opposed to combining all in one `sources.cmake`?

File: src/telemetry/pings.yaml [hunk 2]
```diff
@@ -9,6 +9,19 @@
...
+    A simple ping to send glean events generated by the Mozilla Cuddlefish apps.
+  include_client_id: true
+  send_if_empty: false
+  bugs:
+    - https://bugzilla.mozilla.org/show_bug.cgi?id=1708637
+  data_reviews:
+    - https://bugzilla.mozilla.org/show_bug.cgi?id=1711123
+  notification_emails:
+    - amarchesini@mozilla.com
+    - vpn-telemetry@mozilla.com
```
Comment (Gela) [inline src/telemetry/pings.yaml:23]: Why was this added?

File: src/translations/strings.yaml [hunk 3]
```diff
@@ -852,6 +852,7 @@ inAppMessaging:
   searchBarPlaceholderText: Search messages
   emptyStateTitle: No messages
   emptyStateDescription: You’re up to date!
+  dateTimeYesterday: Yesterday
```
Comment (Gela) [inline src/translations/strings.yaml:855]: wondering also why this was added?

**Review by Gela — COMMENTED** [review]
I have a few questions primarily to better my own understanding. Thanks for taking this on!

File: src/translations/strings.yaml (hunk 3 above)
Comment (flodolo) [reply src/translations/strings.yaml:855]: It's a string from `src/shared/translations/strings.yaml`, which is removed in this PR.

File: src/cmake/shared-sources.cmake (hunk 1 above)
Comment (brizental) [reply src/cmake/shared-sources.cmake:6]: This file is shared between tests and mozillavpn. If I merge the two files I'd have to include mozillavpn-sources in the tests instead. That would require a lot of changes, mostly band aids to get it to work. I think keeping this sources file separate for now is cleaner than that.

File: src/telemetry/pings.yaml (hunk 2 above)
Comment (brizental) [reply src/telemetry/pings.yaml:23]: This is coming from src/shared/telemetry/pings.yaml

**Review by Gela — APPROVED** [review]

**Review by strseb — APPROVED** [review]

File: scripts/utils/generate_ts.sh [hunk 4]
```diff
@@ -62,6 +62,7 @@ for branch in $(git branch -r | grep origin/releases); do
 
   UNFLATTENED=false
 
+  # TODO: Remove this once all branches have been updated to not have the shared/ folder.
```
Comment (lesleyjanenorton) [inline scripts/utils/generate_ts.sh:65]: Do we have a ticket for this? Let's create one if not and add the ticket number here please.

**Review by lesleyjanenorton — APPROVED** [review]
